discovery_engine_core/providers/src/gnews/models.rs

```diff
@@ -0,0 +1,89 @@
+// Copyright 2022 Xayn AG
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, version 3.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+use chrono::{DateTime, Utc};
+
+use reqwest::Url;
+use serde::{Deserialize, Serialize};
+
+use crate::utils::deserialize_null_default;
+
+/// A news article
+#[derive(Clone, Serialize, Deserialize, Debug)]
+pub struct Article {
+    #[serde(deserialize_with = "deserialize_null_default")]
+    pub title: String,
+
+    #[serde(deserialize_with = "deserialize_null_default")]
+    pub description: String,
+
+    #[serde(deserialize_with = "deserialize_null_default")]
+    pub content: String,
+
+    #[serde(deserialize_with = "deserialize_null_default")]
+    pub url: String,
+
+    #[serde(deserialize_with = "deserialize_null_default")]
+    pub image: String,
+
+    #[serde(deserialize_with = "deserialize_null_default")]
+    pub source: Source,
+
+    #[serde(rename(deserialize = "publishedAt"), alias = "date_published")]
+    pub published_at: DateTime<Utc>,
+}
+
+#[derive(Clone, Serialize, Deserialize, Debug, Default)]
+pub struct Source {
+    #[serde(deserialize_with = "deserialize_null_default")]
+    pub name: String,
+
+    #[serde(deserialize_with = "deserialize_null_default")]
+    pub url: String,
+}
+
+/// Query response from the Gnews API
+#[derive(Deserialize, Debug)]
+pub(crate) struct Response {
```

Comment:
🔧 this seems split from the query structs, i'd suggest to either move the query related structs here or this struct there (both would fit in one way or another)

Reply:
see same comment on newscatcher